### DISCUSSION TITLE
Multidimensional data

### DISCUSSION THREAD

**Discussion by PhyAMR** [MilesCranmer/PySR#745] [Q&A] [answered]
I want to use PySR in a dataset, but instead of using just one variable (ie. $f(x) = x^2$) I want to use 2 variable to make a 3 dimensional analysis (ie. $f(x,y) = x^2+y^2$). Has anyone have done enything similar?

**Comment by MilesCranmer** [comment]
Not sure I understand. PySR natively supports multiple input features - see the README example. But maybe you mean something else?

**Reply by PhyAMR** [reply]
> Not sure I understand. PySR natively supports multiple input features - see the README example. But maybe you mean something else?

I want to find a function that depends on three variables (ie $f(x,y,z)$) so I don't know if you are refering to the example that uses naural networks or if you can set directly model.fit(x,y,z)

**Reply by MilesCranmer** [reply, marked as answer]
Yes PySR can do this - see example on https://github.com/MilesCranmer/PySR
